test/suites/container_devices_nic_bridged.sh

```diff
@@ -429,6 +432,81 @@ test_container_devices_nic_bridged() {
         lxc exec "${ctName}" -- udhcpc6 -f -i eth0 -n -q -t5 2>&1 | grep 'IPv6 obtained'
   fi
 
+  # Check that dnsmasq will resolve on the lo
+  # We can't guarantee that the container has an ip6 addr so just check the A
+  # record; we only care about access to dnsmasq here, not the record itself.
+  dig -4 +retry=0 +notcp @192.0.2.1 A "${ctName}.${dnsDomain}" | grep "${ctName}.${dnsDomain}.\\+0.\\+IN.\\+A.\\+192.0.2."
+  dig -6 +retry=0 +notcp @2001:db8::1 A "${ctName}.${dnsDomain}" | grep "${ctName}.${dnsDomain}.\\+0.\\+IN.\\+A.\\+192.0.2."
+  dig -4 +retry=0 +tcp @192.0.2.1 A "${ctName}.${dnsDomain}" | grep "${ctName}.${dnsDomain}.\\+0.\\+IN.\\+A.\\+192.0.2."
+  dig -6 +retry=0 +tcp @2001:db8::1 A "${ctName}.${dnsDomain}" | grep "${ctName}.${dnsDomain}.\\+0.\\+IN.\\+A.\\+192.0.2."
+
+  # Check that dnsmasq will resolve from the bridge
+  # testImage doesn't have dig, so we create a netns with eth0 connected to the
+  # bridge instead
+  ip link add veth_left type veth peer veth_right
+  ip link set veth_left master "${brName}" up
+
+  ip netns add testdns
+  ip link set dev veth_right netns testdns
+
+  ip netns exec testdns ip link set veth_right name eth0
+  ip netns exec testdns ip link set dev eth0 up
+  ip netns exec testdns ip addr add 192.0.2.20/24 dev eth0
+  ip netns exec testdns ip addr add 2001:db8::20/64 dev eth0
+
+  ip addr
+  ip netns exec testdns ip addr
+
+  # Give eth0 a chance to finish duplicate addr detection (ipv6)
```

Comment:
Can't easily use `includes/net.sh: wait_for_dad` here as the device is in a netns; happy to refactor in `includes/net.sh` if needed.